hummingbot/market/kraken/kraken_market.pyx

```diff
@@ -251,6 +251,20 @@ cdef class KrakenMarket(MarketBase):
             return hb_trading_pair
         base = KrakenMarket.convert_to_exchange_symbol(base)
         quote = KrakenMarket.convert_to_exchange_symbol(quote)
+
+        # Exceptions for special trading pairs
+        if (base + quote) in constants.SPECIAL_PAIRS:
+            exchange_trading_pair = f"{base}{delimiter}{quote}"
+            return exchange_trading_pair
+        all_special_symbols = constants.SPECIAL_BASES + constants.SPECIAL_QUOTES
+        if base in all_special_symbols and quote in all_special_symbols:
+            if base in constants.SPECIAL_BASES:
+                base = "X" + base
+            if quote in constants.SPECIAL_QUOTES:
+                quote = "Z" + quote
+            elif quote in constants.SPECIAL_BASES:
+                quote = "X" + quote
+
```

Comment:
are we really sure any of this is needed? all API endpoints support use of altname instead of name, which in the case of pairs containing assets like XETH, XXBT, etc. is the pair without the extra X (so using ETH, XBT, etc.).

Reply:
https://api.kraken.com/0/public/AssetPairs for reference

Reply:
ah I see, we're doing this because our internal data structures (order book) are using exchange asset names (XETH, XXBT, etc.) due to the API responses. I would prefer we change it there (so that they use ETH, XBT, etc.) rather than here, but after looking at the endpoints it seems like that may not be as simple as it should be. If this works then it's fine.

Reply:
I guess this can be marked as resolved now?